lib/secure_headers/headers/content_security_policy_config.rb

```diff
@@ -72,16 +159,12 @@ def append(new_hash)
     end
 
     def to_h
-      self.class.attrs.each_with_object({}) do |key, hash|
-        value = self.send(key)
+      self.class::ATTRS.each_with_object({}) do |key, hash|
+        value = directive_value(key)
         hash[key] = value unless value.nil?
       end
     end
 
-    def dup
-      self.class.new(self.to_h)
```

Comment:
The converting to hash -> reading from hash was removed in favor of the `from_self` method in many places.

There's still a few places where the `from_hash` helper is used, unfortunately.